src/core/Header/HeaderLogo.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+
+import cx from 'classnames';
+import React from 'react';
+
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+
+export type HeaderLogoProps = {
+  /**
+   * Logo shown before the main content.
+   */
+  logo: JSX.Element;
+  /**
+   * Click event handler.
+   * Will update the Logo to have mouse and keyboard interaction if provided.
+   */
+  onClick?: () => void;
+  /**
+   * Children is put on the right of the logo.
+   */
+  children?: React.ReactNode;
+} & CommonProps;
+
+/**
+ * Header Title section
+ * @example
+ * <HeaderLogo logo={<SvgImodelHollow />}>iTwin Application</HeaderLogo>
+ * <HeaderLogo logo={<SvgImodelHollow />} />
+ * <HeaderLogo logo={<img src='image.png' />} />
+ * <HeaderLogo logo={<img src='data:image/png;base64,...' />} >Downloaded Image</HeaderLogo>
+ */
+export const HeaderLogo = (props: HeaderLogoProps) => {
+  const { className, children, logo, onClick, ...rest } = props;
+  const keyDownHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (onClick && (e.key === 'Enter' || e.key === ' ')) {
+      e.preventDefault();
+      onClick();
+    }
+  };
+  useTheme();
+  return (
+    <div
+      className={cx('iui-header-logo', className)}
+      role={onClick ? 'button' : undefined}
+      tabIndex={onClick ? 0 : undefined}
+      onKeyDown={onClick ? keyDownHandler : undefined}
```

Comment:
YOu check onClick double time in here and in function.. Remove one

Reply:
I did not want to have a keyDownHandler if there was no reason for it to be there, but I can leave it always there... And if I dont check it in the function, there is a branch that cannot be reached in the tests.